Sender_C/sender_main.c

```diff
@@ -0,0 +1,121 @@
+#include "Common_header.h"
+#include "time.h"
+#include "Output_Handler.h"
+#include "sender_main.h"
+
+
+
+int main()
+{
+#if(TEST_CODE_ACTIVE != TRUE)
+	clock_t timestamp_start;
+	timestamp_start = clock();
+	sender_main();
+	while(1)
+	{
+		if(Fc_counter_elapsed(timestamp_start) == TRUE)
+		{
+			timestamp_start = clock();
+			if(sender_main() == PROCESS_TERMINATE)
+			{
+				break;
+			}
+		}
+	}
+#else
+	int return_val[5] , i ,j = 0;
+	return_val[0] = Test_OpHdlr_positive_trigger();
+	return_val[1] = Test_OpHdlr_negetive_trigger();
+	return_val[2] = Test_BattParamColl(0,565 , 89);
+	return_val[3] = Test_BattParamColl(0,565 , UPDATE);
+	return_val[4] = Test_BattParamColl(0,565 , ACCUMULATE);
+	for(i = 0 ; i < 5 ; i++)
+	{
+		if(return_val[i] == NOT_OK)
+		{
+			j= 1;
+		}
+	}
+	if(j ==0)
+	{
+		printf("all case passed\n");
+	}
+	else
+	{
+		printf("test failed");
+	}
+	if(Test_CSV_HandlerNoFile() == OK)
+	{
+		printf("No file handling correct\n");
+	}
+	if(Test_CSV_HandlerEndFile() == OK)
+	{
+		printf("Empty/ End of file handling correct");
+	}
+
+	getchar();
+#endif
+	return 0;
+}
+
+int sender_main(void)
+{
+	/*Operation steps:
+	1. Collect a set of parameters from CSV file.
+	2. Store it in Local memory/Msg collect
+	3. Send it to the configured user interface*/
+	int Retval, doTerminate = FALSE;
+	char *disp_msg; 
+	Sender_st Sender = {CSV_Handler , batt_param_collector , Msg_Handler};
+	/*-----------------------------------------*/
+	Retval = Sender.CSVhandler();
+	/*-----------------------------------------*/
+	if(Retval == OK)
+	{
+		disp_msg = Sender.BattParamCol(0 , 0 , ACCUMULATE) ; 
+	}		
+	else if(Retval == NOT_OK)
+	{
+		disp_msg  = "CSV file read error" ;  
+		doTerminate = TRUE;
+	}
+	else if(Retval == FILE_END_OK)
+	{
+		disp_msg  = "CSV file end" ;  
+		doTerminate = TRUE;
+	}
+	/*-----------------------------------------*/
+	Retval = Sender.MsgHndlr(send_to_console , disp_msg);

```

Comment:
Name functions according to their purpose. `MsgHndlr` doesn't express what a consumer of the function could expect it to do.